docs/book/beyond_basics/generating_crd.md

```diff
@@ -0,0 +1,137 @@
+```sh
+# Generate manifests e.g. CRD, RBAC etc.
+manifests:
+	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
+```
+
+When you run `make manifests`, you should see an output like below. CRDs are generated under `configs/crds` directory.
+```sh
+$ make manifests
+go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
+CRD manifests generated under '<curr_dir>/config/crds'
+RBAC manifests generated under '<curr_dir>/config/rbac'
+```
+
+Though `controller-gen` generates manifests for RBAC as well, but this section describes the generation of CRD manifest. 
+
+`controller-gen` reads kubebuilder annotations of the form `// +kubebuilder:something...` defined as Go comments in the `<your-api-kind>_types.go` file under `pkg/apis/...` to produce the CRD manifests. Sections below describe various supported annotations.
+
+## Validation
+
+CRDs support [validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation) by definining ([OpenAPI v3 schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject)) in the validation section. To learn more about the validation feature, refer to the original docs [here](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation). One can specify validation for a field by annotating the field with kubebuilder annotation which is of the form`// +kubebuilder:validation:<key=value>`. If you want to specify multiple validations for a field, you can add multiple such annotation as demonstrated in the example below.
+
+Currently, supporting keys are `Maximum`, `Minimum`, `MaxLength`, `MinLength`, `MaxItems`, `MinItems`, `UniqueItems`, `Enum`, `Pattern`, `ExclusiveMaximum`,
+ `ExclusiveMinimum`, `MultipleOf`, `Format`.  The `// +kubebuilder:validation:Pattern=.+:.+` annotation specifies the Pattern validation requiring that the `Image` field match the regular expression `.+:.+`
+
+**Example:**
+
+```go
+type ToySpec struct {
+
+	// +kubebuilder:validation:Maximum=100
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:ExclusiveMinimum=true
+	Power  float32 `json:"power,omitempty"`
+
+	Bricks int32   `json:"bricks,omitempty"`
+	// +kubebuilder:validation:MaxLength=15
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+
+	// +kubebuilder:validation:MaxItems=500
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:UniqueItems=false
+	Knights []string `json:"knights,omitempty"`
+
+	// +kubebuilder:validation:Enum=Lion,Wolf,Dragon
+	Alias string `json:"alias,omitempty"`
+
+	// +kubebuilder:validation:Enum=1,2,3
+	Rank    int    `json:"rank"`
+}
+
+```
+
+## Additional printer columns
+
+Starting with Kubernetes 1.11, kubectl uses [server-side
+printing](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns).
```

Comment:
It says `server-side printing` but points to `CRD additional printer columns feature`  document.
I'd suggest removing the link in this sentence and moving the link to the later sentence:
```
You can [customize these columns using a CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns).
```

Reply:
Done.